tools/merge-api-changelogs

```diff
@@ -0,0 +1,115 @@
+
+
+def increment_and_get_feature_level() -> int:
+    new_feature_level = None
+    with fileinput.FileInput("version.py", inplace=True) as file:
+        for line in file:
+            if line.startswith("API_FEATURE_LEVEL = "):
+                match = re.search(r"\d+", line)
+                if match:
+                    new_feature_level = int(match.group()) + 1
+                    print(f"API_FEATURE_LEVEL = {new_feature_level}")
+                    continue
+
+            print(line, end="")
+
+    assert new_feature_level is not None
+    print(f"Updated API feature level: {new_feature_level - 1} -> {new_feature_level}")
+    return new_feature_level
+
+
+def merge_changelogs(changelogs: str, new_feature_level: int) -> None:
+    changlogs_merged = False
+    changelog_path = "api_docs/changelog.md"
+    with fileinput.FileInput(changelog_path, inplace=True) as file:
+        for line in file:
+            print(line, end="")
+            if not changlogs_merged and bool(re.fullmatch(r"## Changes in Zulip \d+\.\d+\n", line)):
+                changlogs_merged = True
+                print(f"\n**Feature level {new_feature_level}**\n")
+                print(changelogs, end="")
+
+    print(f"Changelogs merged to {changelog_path}.")
+
+
+def update_feature_level_in_api_docs(new_feature_level: int) -> None:
+    changelog_files_list = get_changelog_files_list()
+    num_replaces = 0
+    api_docs_path = "zerver/openapi/zulip.yaml"
+    with fileinput.FileInput(api_docs_path, inplace=True) as file:
+        for line in file:
+            for file_name in changelog_files_list:
+                old_line = line
+                line = line.replace(file_name, str(new_feature_level))
+                if old_line != line:
+                    num_replaces += 1
+
+            print(line, end="")
+
+    if num_replaces:
+        print(f"Updated {api_docs_path}; {num_replaces} replaces were made.")
+    else:
+        print(f"No changes were made to {api_docs_path}; no matching file name found.")
+
+
+def remove_unmerged_changelog_files() -> None:
+    changelog_files_list = get_changelog_files_list()
+    for file_name in changelog_files_list:
+        os.remove(f"api_docs/unmerged.d/{file_name}")
+
+    if changelog_files_list:
+        print("Removed all the unmerged changelog files.")
+
+
+if __name__ == "__main__":
+    changelogs = get_unmerged_changelogs()
+    if changelogs:
+        new_feature_level = increment_and_get_feature_level()
+        merge_changelogs(changelogs, new_feature_level)
+        update_feature_level_in_api_docs(new_feature_level)
+        remove_unmerged_changelog_files()
+
+        commit_message = f"""changelog: Merge unmerged changelogs and increment API feature level.
+
+This is an automated commit that merges all unmerged changelogs
+and adds them to changelog.md and and increments the API feature level
+to {new_feature_level}."""
+
+        try:
+            subprocess.run(["git", "add", "."], check=True)
```

Comment:
Let's just `git add -u`; that's a lot safer than adding every untracked file in the checkout.